File: FCARDIO.Core/FCARDIO.Core/Extension/ListExtensions.vb
Imports System.Runtime.CompilerServices
Imports System.Text

Namespace Extension
    Public Module ModListExtensions
        <Extension>
        Public Function Join(Of TOuter)(outer As IEnumerable(Of TOuter), Optional ByVal sSplit As String = ",") As String
            If outer.Count = 0 Then Return String.Empty

            Dim sJoin = New StringBuilder(4096)

            For Each o As TOuter In outer
                sJoin.Append(o.ToString())
                sJoin.Append(sSplit)
            Next
            sJoin.Length -= sSplit.Length

            Return sJoin.ToString()
        End Function
    End Module
End Namespace

File: NetStandard/DoNetDrive.Common/DoNetDrive.Common/Extensions/ListExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DoNetDrive.Common.Extensions
{
    /// <summary>
    /// 集合扩展类
    /// </summary>
    public static class ListExtensions
    {

        /// <summary>
        /// 集合扩展，把集合转字符串拼接 参数 分割
        /// </summary>
        /// <typeparam name="TOuter"></typeparam>
        /// <param name="outer"></param>
        /// <param name="sSplit"></param>
        /// <returns></returns>
        public static string Join<TOuter>(this IEnumerable<TOuter> outer, string sSplit = ",")
        {
            if (outer.Count() == 0)
                return string.Empty;

            var sJoin = new StringBuilder(outer.Count() * 10);

            foreach (TOuter o in outer)
            {
                sJoin.Append(o.ToString());
                sJoin.Append(sSplit);
            }
            sJoin.Length -= sSplit.Length;

            return sJoin.ToString();
        }
    }
}
